### DISCUSSION TITLE
Can't add matter device to my Home Assisant - No border router detected

### DISCUSSION THREAD

**Discussion by mforbiz** [openthread/openthread#10065] [Q&A] [answered]
Hi.
I've and OpenThread Border Router installed on an ESP router board.
All seems works fine an I can access to it via web GUI:

![image](https://github.com/openthread/openthread/assets/105542757/8f678222-816d-4dc9-bba2-d987aa0c6d2a)
Than I've tried to add a new matter device to Home Assistant via HA app on my smartphone but after few seconds it say to me that : _`A Thread border router is required - Connect a border router and try again.`_

I've Matter Server installed on my HA:
![image](https://github.com/openthread/openthread/assets/105542757/33fce568-c722-4ac3-abf9-f638c478af86)

and Thread device too:

![image](https://github.com/openthread/openthread/assets/105542757/b257df4f-ca2b-4625-a497-d406a47ccc03)

Here I can see the OTBR and I've tried to import credentials:

![image](https://github.com/openthread/openthread/assets/105542757/b89b0203-3ebe-4293-94f9-7fcd187e4a5f)

But after few second of syncronizzation I see the message: no credential to import

![image](https://github.com/openthread/openthread/assets/105542757/ab48a7bf-0f34-4d21-9f49-267c4aaf8446)

Where am I wrong?

Any idea?

Thansk in advance

**Comment by jwhui** [comment]
With Home Assistant, the Espressif Border Router needs to share its Thread Credentials with the mobile OS - this requires a mobile app that supports the respective iOS or Android APIs:
- https://developer.apple.com/documentation/threadnetwork/
- https://developers.home.google.com/thread

**Reply by mforbiz** [reply]
Hi jwhui.
Thanks for yor reply and sorry for delay in my reply.
I've solved.
If you're interested take a look to this thread on HA forum:
[https://community.home-assistant.io/t/cant-add-otbr-credentials-on-ha/721986](url)
All works fine now.

**Reply by jwhui** [reply, marked as answer]
Thanks for the update. It's as I suspected above - the Thread Credentials need to be shared with the mobile OS, which is detailed in the solution here: https://community.home-assistant.io/t/cant-add-otbr-credentials-on-ha/721986/8

**Comment by roblamoreaux** [comment]
I just went through this nightmare and figured out what causes it. I didn’t do the sync thread credentials to my phone step.

I wish the error message would suggest that you see step three of new networks at
[Thread - Home Assistant](https://www.home-assistant.io/integrations/thread/#turning-home-assistant-into-a-thread-border-router) where it says:

Before you can add Matter-based Thread devices, your phone needs to know the credentials of your newly created Thread network.
To share the credentials with your Android phone, open the Home Assistant Companion app.
In the Companion app, go to Settings > Companion app > Troubleshooting, then select Sync Thread credentials.
Follow the instructions on screen.
Result: You will see a confirmation stating that Thread credentials from Home Assistant have been added to this device.
To share the credentials with your iPhone, open the Home Assistant Companion app.
Go to [Settings > Devices & services](https://my.home-assistant.io/redirect/integrations), select the Thread integration.
Select Configure under Services.
At the bottom of the preferred network box, select Send credentials to phone.

**Reply by mforbiz** [reply]
Hi roblamoreaux
What kind of Matter/Thread device are you using with HA?
At the time of this post I used Matter/thread devices created on the [ESP zero Code](https://zerocode.espressif.com/) platform and using ESP32-H2 boards.
Anyway from many months devices created on this platform don't works too with HA due to a certificate issue.
For this reason I'm stuck with matter/thread developement on HA.

**Reply by roblamoreaux** [reply]
I was playing with it to add the Thread Matter Light ESP32-H2 device using my Android Phone and the HA Companion App. It wouldn't do it with the No Border router detected message. After going in and syncing the credentials it worked fine. I think the commissioning is done by Android not HA, so the Phone needs to have the credentials, but the companion app doesn't do it automatically. I have yet to try any production Matter/Thread devices.

**Reply by roblamoreaux** [reply]
I programmed the ESP32-H2 from https://espressif.github.io/esp-launchpad/?flashConfigURL=https://espressif.github.io/esp-matter/launchpad.toml

**Reply by mforbiz** [reply]
Hi roblamoreaux
Using the link to ESP launchpad I see four different applications, but, unlike the ESP Zero Code platform I can't know (_or I'm not capable to find_) how the board is configured for each of the application:
![image](https://github.com/user-attachments/assets/7e4cee3b-dfff-4fc6-affe-4e8c640062b8)
In other words I can't find anything like this:
![image](https://github.com/user-attachments/assets/4715c8a7-f3a1-4001-8ec2-922df2bd4d63)
I don't know wich are I/O pins used on the ESP32-H2 board.
Also, I'm using[ OpenThread border](https://openthread.io/guides/border-router) router and, if I've well understood, you're not using any border router.
Is this right?

**Comment by roblamoreaux** [comment]
The applications on Launchpad are not configurable. and work for the devkit boards. Since I was testing connecting Matter devices I was fine with that. The code for them are in the espressif esp-matter repository as examples. Now that I know I can connect the devices I will be trying the zerocode configuring

**Reply by mforbiz** [reply]
Ok. Zero code platform is very easy. Le me know if you succesfully create a matter device and it will be recognized without proble in HA.

**Reply by roblamoreaux** [reply]
I just created an ESP32-H2 Matter Thread light fixture using the default setting and added it to Home Assistant without issue. 

Before I told my Home Assistant companion App to Sync Thread Credentials this would not work. It does now.

**Reply by mforbiz** [reply]
Hi.
Have you used any border router ?
